.husky/scripts/commit-msg.js

```diff
@@ -0,0 +1,160 @@
+
+/**
+ * Expected change prefixes, in form of `[predicate, getPrefix]` functions.
+ */
+const changePrefix = [
+    // packages/sdk-core/file.ts => sdk-core
+    // tools/cli/file.ts => cli
+    [oneOf(startsWith('packages/'), startsWith('tools/')), nthPathPart(1)],
+
+    // smoketests/dir/file.ts => smoketests
+    // scripts/dir/file.ts => scripts
+    [oneOf(startsWith('smoketests/'), startsWith('scripts/')), nthPathPart(0)],
+];
+
+/**
+ * Retrieves change prefix from file path, e.g. "sdk-core" for "packages/sdk-core/file.ts".
+ */
+function getChangePrefix(filePath) {
+    for (const [test, prefix] of changePrefix) {
+        if (test(filePath)) {
+            return prefix(filePath);
+        }
+    }
+}
+
+/**
+ * Returns unique values of `values`.
+ */
+function unique(...values) {
+    return [...new Set(...values)];
+}
+
+/**
+ * Returns all expected prefixes for `modifiedFiles`.
+ */
+function getExpectedPrefixes(modifiedFiles) {
+    return unique(modifiedFiles.split('\n').map(getChangePrefix)).filter((f) => !!f);
+}
+
+/**
+ * Returns all used prefixes in `message`.
+ */
+function getActualPrefixes(message) {
+    if (!message.includes(':')) {
+        return [[], message];
+    }
+
+    const [prefix, rest] = message.split(':');
+    if (!prefix) {
+        return [[], message];
+    }
+
+    return [prefix.split(', '), (rest && rest.trim()) || ''];
+}
+
+/**
+ * Returns all prefixes that are in `expected`, but not in `actual`.
+ */
+function getMissingPrefixes(expected, actual) {
+    const missing = [];
+
+    for (const prefix of expected) {
+        if (!actual.includes(prefix)) {
+            missing.push(prefix);
+        }
+    }
+
+    return missing;
+}
+
+/**
+ * Returns normalized expected prefixes based on what is missing and what was provided.
+ * Normalization changes the case to match expected.
+ */
+function normalizeExpectedPrefixes(expected, missing, actual) {
+    const normalizedActual = actual.map((a) => expected.find((e) => ciEquals(a, e)) || a);
+    const missingFromNormalized = missing.filter((m) => !normalizedActual.includes(m));
+    return [...normalizedActual, ...missingFromNormalized];
+}
+
+/**
+ * Displays error banner in format of:
+ * ```
+ * ===============
+ * ==== error ====
+ * ===============
+ * ```
+ */
+function banner(str) {
+    const repeat = (n, what) => [...new Array(n)].map(() => what).join('');
+
+    console.error(chalk.red(repeat(str.length + 10, '=')));
+    console.error(chalk.red(`${repeat(4, '=')} ${str.toUpperCase()} ${repeat(4, '=')}`));
+    console.error(chalk.red(repeat(str.length + 10, '=')));
+}
+
+const message = process.argv[2];
+const modifiedFiles = execSync('git diff --cached --name-only').toString('utf-8');
+const expected = getExpectedPrefixes(modifiedFiles);
+const [actual, rest] = getActualPrefixes(message);
+
+if (!actual.length) {
+    banner('Commit message is invalid - missing prefix');
```

Comment:
still makes sense to add an NFC commit - for example: how you can modify README.md in the root directory?

Reply:
This only will detect changes in `packages/`, `tools/`, `smoketests/` and `scripts/` directories.